test/test_helper.rb

```diff
@@ -106,6 +106,16 @@ def default_index_settings
   {settings: {index: {number_of_shards: 1, number_of_replicas: 0}}}
 end
 
+def default_index_mappings
+  if es_version_5_x?
+    {mappings: {percolator: {properties: {query: { type: "percolator"}}}}}
+  elsif es_version_2_x?
+    {}
+  else
+    raise IncompatibleVersionException "Percolator API not supported for ES #{$client.version}"
+  end
+end
```

Comment:
I don't think we should define default index mappings for all tests in this fashion. The percolator definition only applies to the percolator tests; this code should be moved into the specific test case `setup` block.

Reply:
ah good point will do